scheduler/src/cook/mesos/ranker.clj

```diff
@@ -0,0 +1,170 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns cook.mesos.ranker
+  (:require [clojure.core.async :as async]
+            [clojure.tools.logging :as log]
+            [cook.datomic :as datomic]
+            [cook.mesos.dru :as dru]
+            [cook.mesos.share :as share]
+            [cook.mesos.util :as util]
+            [datomic.api :as d]
+            [metrics.meters :as meters]
+            [metrics.timers :as timers]
+            [plumbing.core :as pc]))
+
+(timers/deftimer [cook-mesos scheduler filter-offensive-jobs-duration])
+
+(defn is-offensive?
```

Comment:
The offensive filter related functions have been moved above the ranking related functions.